libs/blocks/merch/merch.js

```diff
@@ -566,14 +587,21 @@ export async function getCheckoutLinkConfig(
   if (!checkoutLinkConfigs.data.length) return undefined;
   const { locale: { region } } = getConfig();
 
+  // place items with extra options first
+  checkoutLinkConfigs.data.sort((itema, itemb) => {
+    const apf = itema[NAME_PRODUCT_FAMILY];
+    const bpf = itemb[NAME_PRODUCT_FAMILY];
+    return apf.includes('+') && !bpf.includes('+') ? -1 : 1;
+  });
+
   const { paCodeConfigs, productCodeConfigs, productFamilyConfigs } = checkoutLinkConfigs
     .data.reduce(
       (acc, config) => {
-        if (config[NAME_PRODUCT_FAMILY] === paCode) {
+        if (isProductMatch(config, paCode, svar, acc.paCodeConfigs)) {
           acc.paCodeConfigs.push(config);
-        } else if (config[NAME_PRODUCT_FAMILY] === productCode) {
+        } else if (isProductMatch(config, productCode, svar, acc.productCodeConfigs)) {
           acc.productCodeConfigs.push(config);
-        } else if (config[NAME_PRODUCT_FAMILY] === productFamily) {
+        } else if (isProductMatch(config, productFamily, svar, acc.productFamilyConfigs)) {
           acc.productFamilyConfigs.push(config);
         }
```

Comment:
can we rather have one method taking config, acc, PA, product code, and productFamily as entries?
i find code a bit hard to read, because of getting all that conditional logic + abstract method `isProductMatch` with complicated signature in 2 different places.
i feel like having accumulator processed in one place will be easier to understand

Reply:
Done.